src/http/oauth2.ts

```diff
@@ -140,8 +164,8 @@ export function createOAuth2HttpClient(
         } :
         {}
     );
-    await setTokenData(tokenResult.data);
-    return tokenResult.data;
+
+    return await dateAndSetTokenData(tokenResult.data);
```

Comment:
To prevent adding the `creationTimestamp` and extra return statement everywhere in the 3 spots, I've consolidated it in the function which was already used. Happy to discuss if this feels too complex